modules/services/utils/src/test/java/au/com/shiftyjelly/pocketcasts/utils/FileUtilTest.kt

```diff
@@ -0,0 +1,371 @@
+
+        assertEquals("hello", fileName)
+    }
+
+    @Test
+    fun `get file name without extension for a file with multiple dots`() {
+        val file = tempDir.newFile("hello.there..traveller")
+
+        val fileName = FileUtil.getFileNameWithoutExtension(file)
+
+        assertEquals("hello.there.", fileName)
+    }
+
+    @Test
+    fun `get file name without extension for a file with no dots`() {
+        val file = tempDir.newFile("hello")
+
+        val fileName = FileUtil.getFileNameWithoutExtension(file)
+
+        assertEquals("hello", fileName)
+    }
+
+    @Test
+    fun `get file name without extension for a directory with a single dot`() {
+        val dir = tempDir.newFile("hello.there")
+
+        val dirName = FileUtil.getFileNameWithoutExtension(dir)
+
+        assertEquals("hello", dirName)
+    }
+
+    @Test
+    fun `get directory name without extension for a directory with multiple dots`() {
+        val dir = tempDir.newFile("hello.there..traveller")
+
+        val dirName = FileUtil.getFileNameWithoutExtension(dir)
+
+        assertEquals("hello.there.", dirName)
+    }
+
+    @Test
+    fun `get directory name without extension for a directory with no dots`() {
+        val dir = tempDir.newFile("hello")
+
+        val dirName = FileUtil.getFileNameWithoutExtension(dir)
+
+        assertEquals("hello", dirName)
+    }
+
+    @Test
+    fun `compute empty directory size`() {
+        val dir = tempDir.newFolder()
+
+        val size = FileUtil.folderSize(dir)
+
+        assertEquals(0, size)
+    }
+
+    @Test
+    fun `compute directory size with an empty file`() {
+        val dir = tempDir.newFolder().apply {
+            File(this, "file").also(File::createNewFile)
+        }
+
+        val size = FileUtil.folderSize(dir)
+
+        assertEquals(0, size)
+    }
+
+    @Test
+    fun `compute directory size with a non-empty file`() {
+        val dir = tempDir.newFolder().apply {
+            File(this, "file").writeRandomBytes(100)
+        }
+
+        val size = FileUtil.folderSize(dir)
+
+        assertEquals(100, size)
+    }
+
+    @Test
+    fun `compute directory size with complex structure`() {
+        val dir = tempDir.newFolder().apply {
+            File(this, "file1").writeRandomBytes(1)
+            File(this, "file2").writeRandomBytes(10)
+
+            File(this, "inner1").also(File::mkdirs).apply {
+                File(this, "file3").writeRandomBytes(100)
+
+                File(this, "inner2").also(File::mkdirs).apply {
+                    File(this, "file4").writeRandomBytes(1000)
+                    File(this, "file5").writeRandomBytes(10000)
+                }
+            }
+
+            File(this, "inner3").also(File::mkdirs).apply {
+                File(this, "file6").writeRandomBytes(100000)
+                File(this, "file7").writeRandomBytes(1000000)
+            }
+        }
+
+        val size = FileUtil.folderSize(dir)
+
+        assertEquals(1111111, size)
```

Comment:
That's nice how you got the `1111111` total so if the size is wrong it's easy to see where the error happened. 👍